mesonbuild/interpreter/type_checking.py

```diff
@@ -530,9 +530,24 @@ def link_whole_validator(values: T.List[T.Union[StaticLibrary, CustomTarget, Cus
     NATIVE_KW,
 ]
 
+
+def _name_validator(arg: T.Optional[T.Union[str, T.List]]) -> T.Optional[str]:
```

Comment:
Can you add a type argument to T.List?

Reply:
The list has to be empty, so it has no type. If I added one it would be `T.List[T.Any]`, which is what the value implicitly is already, it ends up just making for a longer annotation